pyvista/core/_validation/validate.py

```diff
@@ -260,33 +436,46 @@ def validate_array(
     ...     must_be_sorted=dict(strict=True),
     ...     must_be_in_range=rng,
     ... )
-    array([ 1,  2,  3,  5,  8, 13])
+    (1, 2, 3, 5, 8, 13)
 
     """
-    arr_out = _cast_to_numpy(arr, as_any=as_any, copy=copy)
+    wrapped = _ArrayLikeWrapper(array)
 
-    # Check type
+    # Check dtype
     if must_be_real:
-        check_real(arr_out, name=name)
-    else:
-        try:
-            check_subdtype(arr_out, np.number, name=name)
-        except TypeError as e:
-            raise TypeError(f"{name} must be numeric.") from e
-
+        check_real(wrapped(), name=name)
     if must_have_dtype is not None:
-        check_subdtype(arr_out, must_have_dtype, name=name)
+        check_subdtype(wrapped(), base_dtype=must_have_dtype, name=name)
+
+    if dtype_out not in (None, float, int, bool):
+        return_type = "numpy"
```

Comment:
should probably raise an error here if `dtype` is specified as a numpy type and if `return_type` is not set to `numpy`

Reply:
This has been added in 5cf5deb1b338b50c56e1e735533079e5883b62ce